frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/LoginHandler.java

```diff
@@ -206,8 +205,20 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                         pair("incorrectPasswordCount", updatedIncorrectPasswordCount),
                         pair("attemptNoFailedAt", configurationService.getMaxPasswordRetries()));
 
-                if (incorrectPasswordCount + 1 >= configurationService.getMaxPasswordRetries()) {
-                    LOG.info("User has now exceeded max password retries");
+                if (updatedIncorrectPasswordCount >= configurationService.getMaxPasswordRetries()) {
+                    LOG.info("User has now exceeded max password retries, setting block");
+
+                    codeStorageService.saveBlockedForEmail(
+                            request.getEmail(),
+                            codeBlockedKeyPrefix,
+                            configurationService.getLockoutDuration());
+
+                    if (isReauthJourney) {
```

Comment:
this has moved from when the credentials are valid to when retries have been exceeded?